src/mbgl/util/string_indexer.cpp

```diff
@@ -9,8 +9,10 @@ const std::string empty;
 }
 
 StringIdentity StringIndexer::get(const std::string& string) {
-    MapType& stringToIdentity = getMap();
-    VectorType& identityToString = getVector();
+    std::unique_lock<std::shared_mutex> writerLock(instance().sharedMutex);
```

Comment:
We can just take a read lock here. If we don't find our string in the cache, we release the read lock and take a write lock to do our insertion.

Reply:
Right, I thought about it but then assumed that write contention is quite low and happening mostly in initialization phase. 
I'll change it the way you suggest.
If the string access is proving to be an issue we could switch to a lock-free solution using atomics and spins, so we avoid potential thread priority inversions as well.

Other aspect to consider is the indexer's use of two containers, which is not ideal memory wise - both in terms of size and fragmentation. Considering we're only doing additions, a single vector and linear search could be more L1 cache friendly on a certain scale.